File: ios/PanWalletSDK/ConnectType.swift
//
//  ConnectType.swift
//  PanWalletSDK
//
//  Created by jujien on 05/09/2022.
//

import Foundation

public enum ConnectType: String {
  case connect = "connect"
  case transfer = "transfer"
  case depositToken = "deposit_token"
  case withdrawToken = "withdraw_token"
  case approve = "approve"
  case sendNFT = "send_nft"
  case buyNFT = "buy_nft"
  case sellNFT = "sell_nft"
  case buyBox = "buy_box"
  case unlockBox = "unlock_box"
  case openBox = "open_box"
  case sendBox = "send_box"
  case stakeNFT = "stake_nft"
  case unstakeNFT = "unstake_nft"
  case dismiss = "cancel_transaction"
  case cancelTransaction = "cancel_transaction_non_success"
}

File: android/PanWalletSDK/src/main/java/com/panwallet/sdk/config/ConnectType.kt
package com.panwallet.sdk.config

enum class ConnectType(var type: String) {
    CONNECT("connect"),
    TRANSFER("transfer"),
    APPROVE("approve"),
    DEPOSIT_TOKEN("deposit_token"),
    WITHDRAW_TOKEN("withdraw_token"),
    BUY_NFT("buy_nft"),
    SELL_NFT("sell_nft"),
    SEND_NFT("send_nft"),
    STAKE_NFT("stake_nft"),
    UN_STAKE_NFT("unstake_nft"),
    BUY_BOX("buy_box"),
    UNLOCK_BOX("unlock_box"),
    OPEN_BOX("open_box"),
    SEND_BOX("send_box"),
    DISMISS("cancel_transaction"),
    CANCEL_TRANSACTION("cancel_transaction_non_success")
}